### ISSUE TITLE
Specular from area lights is disproportionately bright

### ISSUE THREAD

**Issue by DGriffin91** [bevyengine/bevy#13327]
## Bevy version Main https://github.com/bevyengine/bevy/commit/dcb8a13b223fbc9425f7af01b3941dd80a229384, 0.13
This isn't a new issue. Not sure how far it goes back, or if there's exists a previous version without it.

Specular reflections from spot/point area lights are disproportionately bright relative to the diffuse response. This often results in extremely bright specular reflections given sufficiently large area lights.

![image](https://github.com/bevyengine/bevy/assets/33357138/80e82c9b-ecc8-4e77-bd2e-4fa434144126)
*This image also displays other specular-related issues in bevy that are not relevant to this issue, let's keep this issue focused, please refer to the blender comparison images below.*

Test setup: I disabled Tonemapping in both blender and bevy. I visually matched the light brightness using just diffuse. The diffuse falloff in bevy doesn't exactly match blender, but it's close. In blender I'm rendering using Cycles with direct light only (no GI).

I apologize in advanced for the amount of comparison images needed. Note that bevy 0 radius is very similar to blender 0 radius across the board, and bevy 1 radius diffuse is similar to blender 1 radius diffuse (though blender wraps around a bit more here). But bevy 1 radius specular is dramatically brighter than blender 1 radius specular.

Bevy 0 radius, diffuse only:
![bevy_0r_diff](https://github.com/bevyengine/bevy/assets/33357138/5ad84d4e-17aa-42e7-9175-6ef6a5a70cf0)

Blender 0 radius, diffuse only:
![blender_0r_diff](https://github.com/bevyengine/bevy/assets/33357138/8ac4fb86-eaec-48c3-8ee7-bd283986cf44)

Bevy 0 radius, specular only:
![bevy_0r_spec](https://github.com/bevyengine/bevy/assets/33357138/cc669d9f-1faa-4d3e-a7d2-cafb1a89c796)

Blender 0 radius, specular only:
![blender_0r_spec](https://github.com/bevyengine/bevy/assets/33357138/3e78fd08-2bb9-4d3c-9b72-a8449ba2e36a)

Bevy 0 radius, both specular and diffuse:
![bevy_0r](https://github.com/bevyengine/bevy/assets/33357138/d82e9771-bcb3-4aef-967d-264fa7a7f0ac)

Blender 0 radius, both specular and diffuse:
![blender_0r](https://github.com/bevyengine/bevy/assets/33357138/093b7845-08c1-4b65-9133-1daa2ab430ee)

Bevy 1 radius, diffuse only:
![bevy_1r_diff](https://github.com/bevyengine/bevy/assets/33357138/337dae1d-b63f-4a34-ac3a-3a799cf495de)

Blender 1 radius, diffuse only:
![blender_1r_diff](https://github.com/bevyengine/bevy/assets/33357138/6e694838-fbda-4ed1-81f0-7cd397d8b5f4)

Bevy 1 radius, specular only:
![bevy_1r_spec](https://github.com/bevyengine/bevy/assets/33357138/ae81a85d-251d-4f73-aa98-41dbbfebf3c9)

Blender 1 radius, specular only:
![blender_1r_spec](https://github.com/bevyengine/bevy/assets/33357138/6b2c009f-dd2f-4349-bc21-b194f9d153ac)

Bevy 1 radius, both specular and diffuse:
![bevy_1r](https://github.com/bevyengine/bevy/assets/33357138/77ce8583-61cf-4bcd-9a27-86b1e5074653)

Blender 1 radius, both specular and diffuse:
![blender_1r](https://github.com/bevyengine/bevy/assets/33357138/7fac16c6-bdfc-424b-9979-85c17d8c1785)

Bevy 1 radius, both specular and diffuse, sharp:
![bevy_1r_sharp](https://github.com/bevyengine/bevy/assets/33357138/2a6a60cf-8160-4885-a25f-d867cbb7727e)

Blender 1 radius, both specular and diffuse, sharp:
![blender_1r_sharp](https://github.com/bevyengine/bevy/assets/33357138/acafd575-8beb-4d70-a170-8c5370125107)

Blender Test Scene (Blender 4.1): [bevy_spec_ref.zip](https://github.com/bevyengine/bevy/files/15281053/bevy_spec_ref.zip)
Bevy test scene (Bevy Main https://github.com/bevyengine/bevy/commit/dcb8a13b223fbc9425f7af01b3941dd80a229384):
```rs
fn main() {
    App::new()
        .insert_resource(ClearColor(Color::BLACK))
        .insert_resource(AmbientLight::NONE)
        .add_plugins(DefaultPlugins)
        .add_systems(Startup, setup)
        .add_systems(Update, screenshot_on_spacebar)
        .run();
}
fn setup(
    mut commands: Commands,
    mut meshes: ResMut<Assets<Mesh>>,
    mut materials: ResMut<Assets<StandardMaterial>>,
) {
    let sphere_mesh = meshes.add(Sphere::new(1.0).mesh().uv(128, 64));
    let surface_material = materials.add(StandardMaterial {
        // Set RGB to 0 to disable diffuse, otherwise use 0.5
        base_color: Color::linear_rgba(0.5, 0.5, 0.5, 1.0),
        // Set reflectance to 0 to disable specular, otherwise use 0.5
        reflectance: 0.5,
        perceptual_roughness: 0.5, // use 0.089 or 0.5
        ..default()
    });
    commands.spawn(PbrBundle {
        mesh: meshes.add(Plane3d::default().mesh().size(100.0, 100.0)),
        material: surface_material.clone(),
        ..default()
    });
    commands.spawn(PbrBundle {
        transform: Transform::from_xyz(-2.5, 1.0, 3.0),
        mesh: sphere_mesh.clone(),
        material: surface_material.clone(),
        ..default()
    });
    commands.spawn(PbrBundle {
        transform: Transform::from_xyz(0.0, 2.5, 3.0),
        mesh: sphere_mesh.clone(),
        material: surface_material.clone(),
        ..default()
    });
    commands.spawn(PointLightBundle {
        point_light: PointLight {
            intensity: 80000.0,
            radius: 1.0,
            ..default()
        },
        transform: Transform::from_xyz(0.0, 1.0, 0.0),
        ..default()
    });
    commands.spawn(Camera3dBundle {
        projection: Projection::Perspective(PerspectiveProjection {
            fov: 45.2f32.to_radians(),
            ..default()
        }),
        tonemapping: Tonemapping::None,
        transform: Transform::from_xyz(0.0, 1.0, 7.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..default()
    });
}
fn screenshot_on_spacebar(
    input: Res<ButtonInput<KeyCode>>,
    main_window: Query<Entity, With<PrimaryWindow>>,
    mut screenshot_manager: ResMut<ScreenshotManager>,
    mut counter: Local<u32>,
) {
    if input.just_pressed(KeyCode::Space) {
        let path = format!("./screenshot-{}.png", *counter);
        *counter += 1;
        screenshot_manager
            .save_screenshot_to_disk(main_window.single(), path)
            .unwrap();
    }
}
```

**Comment by geckoxx** [comment]
You mentioned a "Soft Falloff" Settung in Blender in #13318. Is it enabled in the comparison images?

**Comment by DGriffin91** [comment]
@geckoxx It is not.

**Comment by cart** [comment]
> In blender I'm rendering using Cycles with direct light only (no GI).

Shouldn't we be using Eevee as our "baseline blender comparison", given that Cycles is a raytracer whereas Eevee is structured more like a game engine renderer?

Not a _huge_ deal, as I know Cycles and Eevee are calibrated to each other, but it still seems like Eevee should probably be the point of comparison so its more apples to apples?

**Comment by DGriffin91** [comment]
I did cycles since I figured it would be more accurate and a better "reference" renderer. And eevee makes different tradeoffs than bevy does so comparing to a more accurate renderer that's trying to be generally unbiased is valuable. Valid point though about eevee being more similar to a game engine. I partly didn't want to have even *more* screenshots.

Blender EEVEE 0 radius, diffuse only:
![Blender EEVEE 0 radius, diffuse only](https://github.com/bevyengine/bevy/assets/33357138/e128ce2f-b942-4a5c-b44c-c7af7eb703d2)

Blender EEVEE 0 radius, specular only:
![Blender EEVEE 0 radius, specular only](https://github.com/bevyengine/bevy/assets/33357138/2d4c101c-3712-4da8-b3a2-b300792ccf55)

Blender EEVEE 0 radius, both specular and diffuse:
![Blender EEVEE 0 radius, both specular and diffuse](https://github.com/bevyengine/bevy/assets/33357138/f0ee0195-1833-4206-9a27-147541d0fded)

Blender EEVEE 1 radius, diffuse only:
![Blender EEVEE 1 radius, diffuse only](https://github.com/bevyengine/bevy/assets/33357138/0d38bf24-1532-41b9-b706-606adc34da5d)

Blender EEVEE 1 radius, specular only:
![Blender EEVEE 1 radius, specular only](https://github.com/bevyengine/bevy/assets/33357138/0f6bf9f8-caaf-41b0-a336-e14b4acd589e)

Blender EEVEE 1 radius, both specular and diffuse:
![Blender EEVEE 1 radius, both specular and diffuse](https://github.com/bevyengine/bevy/assets/33357138/4cd30925-7a7a-4f39-af93-2235bece3ca1)

Blender EEVEE 1 radius, both specular and diffuse, sharp:
![Blender EEVEE 1 radius, both specular and diffuse, sharp](https://github.com/bevyengine/bevy/assets/33357138/0e5b5fc5-4621-4f20-8cb0-c96f89803236)

**Comment by cart** [comment]
> I used cycles since I figured it would be more accurate and a better "reference" renderer. And eevee makes different tradeoffs than bevy does so comparing to a more accurate renderer that's trying to be generally unbiased is valuable. Valid point though about eevee being more similar to a game engine. I partly didn't want to have even more screenshots.

Yeah this is a reasonable take. Both seem valuable, but when it comes to calibrating values, I think it makes the most sense to target Eevee, as in theory the way light behaves is more aligned. And when people are designing their scenes in Blender, we have the most hope (and therefore the most value) in having parity with Eevee. We can't really hope to synchronize perfectly with Cycles. And calibrating to Cycles introduces the risk of not syncing correctly with Eevee.

In short: I think we should focus our time on Eevee calibration, except for areas where techniques significantly diverge, or in cases where we're introducing raytracing techniques that are closer to Cycles than whatever Eevee is doing.

**Comment by cart** [comment]
Thanks for the Eeevee shots! It does appear that we're way out of sync with _both_ renderers :)

**Comment by DGriffin91** [comment]
Worth noting that Eevee is a bit more of a moving target. In the next version of blender it will also use ray tracing. 

Our BRDF implementation is much more similar to the one cycles uses. I don't really think we should chase blender in particular, it's just one dcc and certainly not the most popular in the industry. Blender also has bugs and isn't always correct. But cycles is more likely to be correct than Eevee. Ultimately when making comparisons and working on Bevy's renderer, it takes actually understanding the BRDF/BSDF and how the lighting should really work. This comparison specifically sets up cycles to narrow in on the BRDF. It would obviously be easy to accidentally use cycle's GI and be comparing apples to bananas. Using Eevee or cycles as a reference here is just an easy way point out that we're *way* off.

To see one aspect of how our BRDF impl is more similar to Cycles, compare the "0 radius, specular only":

Bevy 0 radius, specular only:
![Bevy 0 radius, specular only](https://github.com/bevyengine/bevy/assets/33357138/6002a7eb-57c9-486c-abaa-131463873186)

Blender 0 radius, specular only:
![Blender 0 radius, specular only](https://github.com/bevyengine/bevy/assets/33357138/3f7c7545-b0a0-403a-815c-208e85eeaa19)

Blender EEVEE 0 radius, specular only
![Blender EEVEE 0 radius, specular only](https://github.com/bevyengine/bevy/assets/33357138/75bc3288-7bf8-4305-a6e5-5e3bd8c04b4b)

**Comment by cart** [comment]
Hmmm solid points. I guess the takeaway here is "find the reference that is appropriate for the algorithm being tested", which is sadly harder than "always use Eevee as a reference" or "always use Cycles as a reference".
